### ISSUE TITLE
Succeeds in connecting, but GUI stuck in "Connecting"

### ISSUE THREAD

**Issue by icodeiexist** [yuezk/GlobalProtect-openconnect#62]
It's seems to be working fine, but the GUI gets stuck in "Connecting", which won't minimize, nor let me switch gateways. Thank you.  
```
Build globalprotect-openconnect_1.3.0-1ppa1_amd64.deb on:
NAME="Linux Mint"
VERSION="20.1 (Ulyssa)"
OpenConnect version v8.10-170-gca7bc365
Using OpenSSL 1.1.1f  31 Mar 2020. Features present: TPM (OpenSSL ENGINE not present), PKCS#11, HOTP software token, TOTP software token, DTLS, ESP
Supported protocols: anyconnect (default), nc, gp, pulse
```

**Comment by CambodianCoder** [comment]
Encountering the same issue currently.

**Comment by icodeiexist** [comment]
Looks like this is a duplicate of https://github.com/yuezk/GlobalProtect-openconnect/issues/43

**Comment by yuezk** [comment]
Let me know if you guys still encounter this problem, I'm going to fix it but lacking sufficient information, it could be better if you could post the logs here.

**Comment by dmitriy-geels** [comment]
I face the same issue. Ubuntu 20.04.3, gpclient 1.3.4+snapshot19.gb0947bf-ppa1~ubuntu20.04.

I think this line might be a clue: https://github.com/yuezk/GlobalProtect-openconnect/blob/0e4e082594315e5d51e5facdee61e3d55c4545f9/GPService/gpservice.cpp#L184

This is the only place I could find, where service changes state to connected.

Here is a piece of log:
```
2021-12-16 18:26:58.646 INFO  [103258] [GPClient::onGatewaySuccess@373] Gateway login succeeded, got the cookie authcookie=cookie-value&portal=gw-internal-portal&user=email%40address&domain=%2528empty_domain%2529&preferred-ip=&computer=dmig-office
2021-12-16 18:26:58.656 INFO  [103258] [GPClient::onVPNLogAvailable@489] Output of `openconnect --version`: OpenConnect version v8.10-3+git20211209-1~ubuntu20.04
Using GnuTLS 3.6.13. Features present: TPMv2, PKCS#11, RSA software token, HOTP software token, TOTP software token, Yubikey OATH, System keys, DTLS, ESP
Supported protocols: anyconnect (default), nc, gp, pulse, f5, fortinet, array
2021-12-16 18:26:58.656 INFO  [103258] [GPClient::onVPNLogAvailable@489] Start process with arugments: --protocol=gp -u  --cookie-on-stdin gwi-sig.hub.domain.net
2021-12-16 18:26:58.656 INFO  [103258] [GPClient::onVPNLogAvailable@489] Openconnect started successfully, PID=103337
2021-12-16 18:26:58.662 INFO  [103258] [GPClient::onVPNLogAvailable@489] POST https://gwi-sig.hub.domain.net/ssl-vpn/getconfig.esp
2021-12-16 18:26:58.953 INFO  [103258] [GPClient::onVPNLogAvailable@489] Connected to [2406:..:..:..:..:..:..:1337]:443
2021-12-16 18:26:58.967 INFO  [103258] [GPClient::onVPNLogAvailable@489] SSL negotiation with gwi-sig.hub.domain.net
2021-12-16 18:26:59.553 INFO  [103258] [GPClient::onVPNLogAvailable@489] Connected to HTTPS on gwi-sig.hub.domain.net with ciphersuite (TLS1.2)-(RSA)-(AES-256-GCM)
2021-12-16 18:26:59.883 INFO  [103258] [GPClient::onVPNLogAvailable@489] Tunnel timeout (rekey interval) is 360 minutes.
Idle timeout is 180 minutes.
2021-12-16 18:26:59.883 INFO  [103258] [GPClient::onVPNLogAvailable@489] GlobalProtect IPv6 support is experimental. Please report results to <openconnect-devel@lists.infradead.org>.
No MTU received. Calculated 1406 for ESP tunnel
2021-12-16 18:26:59.883 INFO  [103258] [GPClient::onVPNLogAvailable@489] POST https://gwi-sig.hub.domain/ssl-vpn/hipreportcheck.esp
2021-12-16 18:27:00.195 INFO  [103258] [GPClient::onVPNLogAvailable@489] WARNING: Server asked us to submit HIP report with md5sum 67cc874fdc2e95cc87ebc2951e380025.
    VPN connectivity may be disabled or limited without HIP report submission.
    You need to provide a --csd-wrapper argument with the HIP report submission script.
2021-12-16 18:27:00.473 INFO  [103258] [GPClient::onVPNLogAvailable@489] ESP session established with server
2021-12-16 18:27:00.473 INFO  [103258] [GPClient::onVPNLogAvailable@489] EPOLL_CTL_DEL: No such file or directory
2021-12-16 18:27:00.474 INFO  [103258] [GPClient::onVPNLogAvailable@489] ESP tunnel connected; exiting HTTPS mainloop.
2021-12-16 18:27:00.475 INFO  [103258] [GPClient::onVPNLogAvailable@489] Configured as 10.224.1.79 + 2406:da18:93e:1342::a:132a, with SSL disconnected and ESP established
2021-12-16 18:27:00.475 INFO  [103258] [GPClient::onVPNLogAvailable@489] Session authentication will expire at Sun Dec 26 18:26:58 2021
```

The reason IMO is: `openconnect` doesn't say 'Connected as' in case of warning albeit the connection is established.

**Comment by dmitriy-geels** [comment]
Another try, this time with HIP trojan:
```
...
2021-12-16 19:24:41.388 INFO  [107542] [GPClient::onVPNLogAvailable@489] Trying to run HIP Trojan script '/usr/libexec/openconnect/hipreport.sh'.
2021-12-16 19:24:41.425 INFO  [107542] [GPClient::onVPNLogAvailable@489] HIP script '/usr/libexec/openconnect/hipreport.sh' completed successfully (report is 2384 bytes).
2021-12-16 19:24:41.426 INFO  [107542] [GPClient::onVPNLogAvailable@489] POST https://gwi-sig.hub.domain.net/ssl-vpn/hipreport.esp
2021-12-16 19:24:41.736 INFO  [107542] [GPClient::onVPNLogAvailable@489] HIP report submitted successfully.
2021-12-16 19:24:42.028 INFO  [107542] [GPClient::onVPNLogAvailable@489] ESP session established with server
2021-12-16 19:24:42.028 INFO  [107542] [GPClient::onVPNLogAvailable@489] EPOLL_CTL_DEL: No such file or directory
2021-12-16 19:24:42.063 INFO  [107542] [GPClient::onVPNLogAvailable@489] ESP tunnel connected; exiting HTTPS mainloop.
2021-12-16 19:24:42.063 INFO  [107542] [GPClient::onVPNLogAvailable@489] Configured as 10.224.1.84 + 2406:da18:93e:1342::a:132f, with SSL disconnected and ESP established
2021-12-16 19:24:42.063 INFO  [107542] [GPClient::onVPNLogAvailable@489] Session authentication will expire at Sun Dec 26 19:24:40 2021
```

Still no changes. Actually, searching for 'Connected as' in openconnect repository shows no results.
You show chose another way of detecting a successful connection.

Here is a string you're looking for: https://gitlab.com/openconnect/openconnect/-/blob/master/main.c#L1479
Changed to 'Configured as' about a year ago -- this is the reason.

**Comment by dmitriy-geels** [comment]
As a temporary workaround, I downgraded openconnect from 8.10-3+git20211209-1~ubuntu20.04 to 8.05-1 from the official repo.

**Comment by kyechou** [comment]
The issue should be resolved in the merged PR #124.
